_ml-commons-plugin/semantic-search.md

```diff
@@ -0,0 +1,1039 @@
+          "model_content_hash_value": "acdc81b652b83121f914c5912ae27c0fca8fabf270e6f191ace6979a19830413",
+          "model_config": {
+            "all_config": """{"_name_or_path":"old_models/msmarco-distilbert-base-tas-b/0_Transformer","activation":"gelu","architectures":["DistilBertModel"],"attention_dropout":0.1,"dim":768,"dropout":0.1,"hidden_dim":3072,"initializer_range":0.02,"max_position_embeddings":512,"model_type":"distilbert","n_heads":12,"n_layers":6,"pad_token_id":0,"qa_dropout":0.1,"seq_classif_dropout":0.2,"sinusoidal_pos_embds":false,"tie_weights_":true,"transformers_version":"4.7.0","vocab_size":30522}""",
+            "model_type": "distilbert",
+            "embedding_dimension": 768,
+            "framework_type": "SENTENCE_TRANSFORMERS"
+          },
+          "last_updated_time": 1694358499122,
+          "last_registered_time": 1694358499121,
+          "name": "huggingface/sentence-transformers/msmarco-distilbert-base-tas-b",
+          "model_group_id": "Z1eQf4oB5Vm0Tdw8EIP2",
+          "model_content_size_in_bytes": 266352827,
+          "algorithm": "TEXT_EMBEDDING"
+        }
+      }
+    ]
+  }
+}
+```
+
+The response contains the model information, including the `model_state` (`REGISTERED`) and the number of chunks it was split into `total_chunks` (27).
+</details>
+
+#### Advanced: Registering a custom model
+
+To register a custom model, you must provide a model configuration in the register request. For example, the following is a register request with the full format for the model used in this tutorial:
+
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "huggingface/sentence-transformers/msmarco-distilbert-base-tas-b",
+  "version": "1.0.1",
+  "model_group_id": "Z1eQf4oB5Vm0Tdw8EIP2",
+  "description": "This is a port of the DistilBert TAS-B Model to sentence-transformers model: It maps sentences & paragraphs to a 768 dimensional dense vector space and is optimized for the task of semantic search.",
+  "model_task_type": "TEXT_EMBEDDING",
+  "model_format": "TORCH_SCRIPT",
+  "model_content_size_in_bytes": 266352827,
+  "model_content_hash_value": "acdc81b652b83121f914c5912ae27c0fca8fabf270e6f191ace6979a19830413",
+  "model_config": {
+    "model_type": "distilbert",
+    "embedding_dimension": 768,
+    "framework_type": "sentence_transformers",
+    "all_config": """{"_name_or_path":"old_models/msmarco-distilbert-base-tas-b/0_Transformer","activation":"gelu","architectures":["DistilBertModel"],"attention_dropout":0.1,"dim":768,"dropout":0.1,"hidden_dim":3072,"initializer_range":0.02,"max_position_embeddings":512,"model_type":"distilbert","n_heads":12,"n_layers":6,"pad_token_id":0,"qa_dropout":0.1,"seq_classif_dropout":0.2,"sinusoidal_pos_embds":false,"tie_weights_":true,"transformers_version":"4.7.0","vocab_size":30522}"""
+  },
+  "created_time": 1676073973126
+}
+```
+
+For more information, see [Model serving framework]({{site.url}}{{site.baseurl}}/ml-commons-plugin/ml-framework/).
```

Comment:
We renamed model serving framework as ML framework. How about change to ML framework?